lib/datasource/git-refs/index.ts

```diff
@@ -1,110 +1,69 @@
-import simpleGit from 'simple-git';
-import * as packageCache from '../../util/cache/package';
-import { getRemoteUrlWithToken } from '../../util/git/url';
+import { cache } from '../../util/cache/package/decorator';
 import * as semver from '../../versioning/semver';
 import type { DigestConfig, GetReleasesConfig, ReleaseResult } from '../types';
+import { GitDatasource } from './base';
 import type { RawRefs } from './types';
 
-export const id = 'git-refs';
-export const customRegistrySupport = false;
-
-const cacheMinutes = 10;
-
 // git will prompt for known hosts or passwords, unless we activate BatchMode
 process.env.GIT_SSH_COMMAND = 'ssh -o BatchMode=yes';
 
-export async function getRawRefs(
-  { lookupName }: GetReleasesConfig,
-  hostType: string
-): Promise<RawRefs[] | null> {
-  const git = simpleGit();
-  const cacheNamespace = 'git-raw-refs';
+export class GitRefsDatasource extends GitDatasource {
+  static readonly id = 'git-refs';
 
-  const cachedResult = await packageCache.get<RawRefs[]>(
-    cacheNamespace,
-    lookupName
-  );
-  /* istanbul ignore next line */
-  if (cachedResult) {
-    return cachedResult;
-  }
-
-  // fetch remote tags
-  const lsRemote = await git.listRemote([
-    getRemoteUrlWithToken(lookupName, hostType),
-  ]);
-  if (!lsRemote) {
-    return null;
+  constructor() {
+    super(GitRefsDatasource.id);
   }
 
-  const refMatch = /(?<hash>.*?)\s+refs\/(?<type>.*?)\/(?<value>.*)/;
-  const headMatch = /(?<hash>.*?)\s+HEAD/;
+  override readonly customRegistrySupport = false;
 
-  const refs = lsRemote
-    .trim()
-    .split('\n')
-    .map((line) => line.trim())
-    .map((line) => {
-      let match = refMatch.exec(line);
-      if (match) {
-        return {
-          type: match.groups.type,
-          value: match.groups.value,
-          hash: match.groups.hash,
-        };
-      }
-      match = headMatch.exec(line);
-      if (match) {
-        return {
-          type: '',
-          value: 'HEAD',
-          hash: match.groups.hash,
-        };
-      }
-      // istanbul ignore next
-      return null;
-    })
-    .filter(Boolean)
-    .filter((ref) => ref.type !== 'pull' && !ref.value.endsWith('^{}'));
-  await packageCache.set(cacheNamespace, lookupName, refs, cacheMinutes);
-  return refs;
-}
-
-export async function getReleases({
-  lookupName,
-}: GetReleasesConfig): Promise<ReleaseResult | null> {
-  const rawRefs: RawRefs[] = await getRawRefs({ lookupName }, id);
+  @cache({
+    namespace: `datasource-${GitRefsDatasource.id}`,
+    key: ({ lookupName }: GetReleasesConfig) => lookupName,
+  })
+  // eslint-disable-next-line class-methods-use-this
+  async getReleases({
```

Comment:
can we add the override keyword? we can validate it later, when all other classes are prepared too.